riak/codecs/pbuf.py

```diff
@@ -605,17 +608,26 @@ def decode_map_value(self, entries):
                 value = entry.flag_value
             elif dtype == 'map':
                 value = self.decode_map_value(entry.map_value)
+            else:
+                raise ValueError(
+                    'Map may not contain datatype: {}'
+                    .format(dtype))
             out[(name, dtype)] = value
         return out
 
     def decode_set_value(self, set_value):
         return [bytes_to_str(string[:]) for string in set_value]
 
+    def decode_hll_value(self, hll_value):
+        return int(hll_value)
```

Comment:
Note: in Python 2, `int()` thankfully does the right thing if `hll_value` requires a `long` internally.